app/Http/Controllers/GroupCreateController.php

```diff
@@ -37,20 +41,80 @@ public function store(Request $req)
         for ($i = 0; $i < sizeof($req->groupMembers); $i++) {
             $groupMember = new GroupMember;
             $groupMember->group_id = $group->id;
-            $groupMember->username = $req->groupMembers[$i]->username;
+            $groupMember->username = $req->groupMembers[$i];
             $groupMember->status_code = GroupsStatusCodeEnum::PENDING; // have to change 'pending' into a constant
             $groupMember->action_group_id = $group->id;
             $groupMember->is_Admin = 0;
             $groupMember->save();
         }
-        return; // need to return somthing
+        return redirect()->action('GroupCreateController@index');
+    }
+
+    public function deleteGroup(Request $req) {
+        if (Group::find($req->modal_groupid_delete)) {
+            $group = Group::find($req->modal_groupid_delete);
```

Comment:
for these you should just check the `$group != null` or just catch the `ModelNotFoundException` as the current way is two queries